scripts/version-sync.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+function usage {
+    echo "Usage: $(basename "$0") [-c] -f FILE_TO_CHANGE REPLACEMENT_FORMAT [-f FILE_TO_CHANGE REPLACEMENT_FORMAT ...]" 2>&1
+    echo 'Synchronize files to latest version in source file'
+    echo '   -s              Specifies source file for version (default is CHANGELOG.md)'
+    echo '   -f              Specifies a file to change and the format for searching and replacing versions'
+    echo '                       FILE_TO_CHANGE is the file to be updated/checked for updates'
+    echo '                       REPLACEMENT_FORMAT is one of (semver, release, api-release)'
+    echo '                           semver indicates to look for a full semver version and replace with the latest full version'
+    echo '                           release indicates to look for a release semver version (x.x.x) and replace with the latest release version'
+    echo '                           api-release indicates to look for a release semver version in the context of an api route and replace with the latest release version'
```

Comment:
Why complexity of 3 REPLACEMENT_FORMATs if the Makefile only cares about `api-release`?

Reply:
ah, i see that it is also using `release` (but not `semver`)?

i guess it doesn't hurt to have an extra for future proofing 😅

Reply:
My plan is to use this script for syncing in other repos as well, and if I update `unstructured` and `unstructured-api-tools` to use this script those will both use `semver`.